core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -114,6 +114,9 @@ public class RESTSessionCatalog extends BaseViewSessionCatalog
   private static final String DEFAULT_FILE_IO_IMPL = "org.apache.iceberg.io.ResolvingFileIO";
   private static final String REST_METRICS_REPORTING_ENABLED = "rest-metrics-reporting-enabled";
   private static final String REST_SNAPSHOT_LOADING_MODE = "snapshot-loading-mode";
+  public static final String REST_PAGINATION_ENABLED = "rest-pagination-enabled";
```

Comment:
I don't feel like we need to enable/disable this.  If a page size is set, it should use that along with the paginated request, otherwise just default to all results and no pagination.

Reply:
(just noticed others made the same comments below)

Reply:
Thanks @danielcweeks for taking a look. I have removed the `REST_PAGINATION_ENABLED` but I think we still need a configuration for the `REST_PAGE_SIZE`. Let me know if you also think we do not require this configuration.

Reply:
@rdblue  @danielcweeks we had a sync, and just wanted to confirm if we wanted to have this `REST_PAGE_SIZE` as configurable from the client side or have the server return back an optimal number for what `pageSize` should be?